### PR TITLE
Fixes issues #1 #7 and #8

### PR DESCRIPTION
**Pull request by jewesta**

### CODE REVIEW

File: src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java [hunk 1]
```diff
@@ -83,21 +85,15 @@ protected BlockElement gateKeep(BlockElement element) {
 	// Lists
 
 	public OrderedListBuilder<B> startOrderedList() {
-		final int indentationLevel = getElement().getIndentationLevel();
-		final OrderedList list = new OrderedList(indentationLevel);
-		return new OrderedListBuilder<>(list, this::append);
+		return new OrderedListBuilder<>(new OrderedList(), this::append);
```
Comment (AndreaUhl90) [inline src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java:88]: Just for confirmation: Are these changes correct? Or are the three removed lines the new changes and the supposed changed one line was the old one? (I am thinking about the merge conflict before)

File: src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java [hunk 2]
```diff
@@ -83,21 +85,15 @@ protected BlockElement gateKeep(BlockElement element) {
...
 	}
 
 	public UnorderedListBuilder<B> startUnorderedList() {
-		final int indentationLevel = getElement().getIndentationLevel();
-		final UnorderedList list = new UnorderedList(indentationLevel);
-		return new UnorderedListBuilder<>(list, this::append);
+		return new UnorderedListBuilder<>(new UnorderedList(), this::append);
```
Comment (AndreaUhl90) [inline src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java:92]: Just for confirmation: Are these changes correct? Or are the three removed lines the new changes and the supposed changed one line was the old one? (I am thinking about the merge conflict before)

File: src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java [hunk 3]
```diff
@@ -83,21 +85,15 @@ protected BlockElement gateKeep(BlockElement element) {
...
 	}
 
 	public TaskListBuilder<B> startTaskList() {
-		final int indentationLevel = getElement().getIndentationLevel();
-		final TaskList list = new TaskList(indentationLevel);
-		return new TaskListBuilder<>(list, this::append);
+		return new TaskListBuilder<>(new TaskList(), this::append);
```
Comment (AndreaUhl90) [inline src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java:96]: Just for confirmation: Are these changes correct? Or are the three removed lines the new changes and the supposed changed one line was the old one? (I am thinking about the merge conflict before)

File: src/main/java/de/relimit/commons/markdown/blockelement/MarkedBlockElementNode.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * Please note that the first element added to this
+ * {@link MarkedBlockElementNode} needs to be a {@link Paragraph}. A marked
+ * block element node cannot start with other {@link BlockElement}s like
+ * {@link Table}s, {@link Image}s or even other {@link AbstractList}s.
+ */
+public abstract class MarkedBlockElementNode extends BlockElementNode {
+
+	protected MarkedBlockElementNode() {
+		super(1);
```
Comment (AndreaUhl90) [inline src/main/java/de/relimit/commons/markdown/blockelement/MarkedBlockElementNode.java:41]: Is this trivial or should there be a comment that explains the `1` in the `super()` method call?

File: src/main/java/de/relimit/commons/markdown/blockelement/admonition/Admonition.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		return display != null ? display : DEFAULT_DISPLAY;
+	}
+
+	public void setDisplay(Display display) {
+		this.display = display;
+		invalidateSerialized();
+	}
+
+	@Override
+	protected String getMarker(MarkdownSerializationOptions options) {
```
Comment (AndreaUhl90) [inline src/main/java/de/relimit/commons/markdown/blockelement/admonition/Admonition.java:64]: I don´t really understand this method so I cannot say if everything is correct here. How would this string look like if everything is set Default? 
"!!! DefaultType.NOTEDisplay.BLOCK(null) String(directive)"

**Review by AndreaUhl90 — COMMENTED** [review]
I´ve reviewed the changes. I often found methods where the `indentationLevel` has been removed as an argument. Is this the latest state (I am just thinking of the merge conflict before). I commented this in one case but did not comment on each one.

File: src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java (hunk 1 above)
Comment (jewesta) [reply src/main/java/de/relimit/commons/markdown/blockelement/BlockElementNodeBuilder.java:88]: They are correct.

I got confused by how the indentation level worked. When I first implemented this I had in mind that every element was responsible for all the indentation. So I assumed with nodes starting at 0 (no indentation) a child on level 5 would add 5 times the indentation (5 x four spaces). However, this paradigm changed during development without me noticing.

Every parent element in fact creates an indentation which is automatically added to every line of child element output. In other words children only need to add 1 (one) indentation if they need to be further indented in relation to their parent. Or noting if they need to remain on their parent's level.

This means that the child indentation level ist always hardcoded to either 0 (zero) or 1.

It is possible to set an indentation of say 3 on any node. E.g. the `Document` which is often the root node. This works: The whole document is moved from level 0 to level 3 (3 x four spaces) and the whole tree moves with it. It is questionable if that makes sense but I left the possibility to increase the indentation in the code.

If however an indentation level is set on any node, the indentation levels of all children are reset to default (0 / 1). I am not sure if that makes sense. But I did not want to change that for now.

File: src/main/java/de/relimit/commons/markdown/blockelement/admonition/Admonition.java (hunk 5 above)
Comment (jewesta) [reply src/main/java/de/relimit/commons/markdown/blockelement/admonition/Admonition.java:64]: I created a unit test for `Admonition` which might clear things up.
https://github.com/relimitgmbh/markdown-builder/blob/47c3e0689dcf22018e1b4e3ef4dc22e5f144dbab/src/test/java/de/relimit/commons/markdown/blockelement/BlockElementTests.java#L41

Also have a look at the specs by MkDocs:
https://squidfunk.github.io/mkdocs-material/reference/admonitions/#changing-the-title

File: src/main/java/de/relimit/commons/markdown/blockelement/MarkedBlockElementNode.java (hunk 4 above)
Comment (jewesta) [reply src/main/java/de/relimit/commons/markdown/blockelement/MarkedBlockElementNode.java:41]: No this is not trivial and you are right. I have added a comment. Have a look and see if this clears things up.

**Pull request merged**